patientsearch/models/sync.py

```diff
@@ -230,25 +262,34 @@ def patient_as_search_params(patient):
     return search_params
 
 
-def internal_patient_search(token, patient):
+def internal_patient_search(token, patient, active_only=False):
     """Look up given patient from "internal" HAPI store, returns bundle"""
-    params = patient_as_search_params(patient)
+    params = patient_as_search_params(patient, active_only)
+
     return HAPI_request(
         token=token, method="GET", resource_type="Patient", params=params
     )
 
 
-def new_resource_hook(resource):
+def new_resource_hook(resource, consider_active=False, reactivate_patient=False):
     """Return modified version of resourse as per new resource rules
 
     Products occasionally require customization of resources on creation.
     This hook manages such, using environment to specialize.
 
     :returns: modified resource
     """
-    if resource.get("id"):
-        # not a new resource, bail
-        return resource
+    if consider_active:
+        if reactivate_patient:
+            if resource.get("id"):
+                return resource
+        else:
+            if resource.get("id") and resource.get("active") is True:
+                return resource
+    else:
+        if resource.get("id"):
+            # not a new resource, bail
```

Comment:
can you add similar comments on the 2 return statements above, explaining some of the complicated context.